plutus-ir/src/Language/PlutusIR/Transform/ThunkRecursions.hs

```diff
@@ -108,6 +111,9 @@ left with the PLC subset. This would be quite elegant - except that the final "l
 would be quite ugly as it would just have to fail if it saw any PIR features.
 -}
 
+thunkRecursionsTerm :: MonadQuote m => Term TyName Name a -> m (Term TyName Name a)
+thunkRecursionsTerm = PLC.rename >=> thunkRecursionsTermGo
```

Comment:
Ditto. I think this *does* need renaming, but it should be done by the caller.

However, at the moment you're effectively running it twice in a row, which is at least wasteful.

Reply:
Plus then you don't have to rename the other perfectly well-named function ;)